src/templates/finops-hub/modules/dataFactory.bicep

```diff
@@ -4863,6 +4939,401 @@ resource pipeline_RerunETL 'Microsoft.DataFactory/factories/pipelines@2018-06-01
   }
 }
 
+//------------------------------------------------------------------------------
+// recommendations export pipeline
+// Triggered by dailyRecommendations trigger
+//------------------------------------------------------------------------------
+@description('Extracts Azure Advisor and custom recommendations from the Resource Graph API.')
+resource pipeline_ExecuteRecommendations 'Microsoft.DataFactory/factories/pipelines@2018-06-01' = {
+  name: '${recommendationsDataSet}_Execute'
+  parent: dataFactory
+  properties: {
+    activities: [
+      { // Set blob timestamp
+        name: 'Set Blob Timestamp'
+        type: 'SetVariable'
+        dependsOn: []
+        policy: {
+          secureOutput: false
+          secureInput: false
+        }
+        userProperties: []
+        typeProperties: {
+          variableName: 'blobExportTimestamp'
+          value: {
+            value: '@concat(utcNow(\'yyyy\'),\'/\',utcNow(\'MM\'),\'/\',utcNow(\'dd\'),\'/\')'
+            type: 'Expression'
+          }
+        }
+      }
+      { // Set instance id
+        name: 'Set Instance Id'
+        type: 'SetVariable'
+        dependsOn: []
+        policy: {
+          secureOutput: false
+          secureInput: false
+        }
+        userProperties: []
+        typeProperties: {
+          variableName: 'instanceId'
+          value: {
+            value: '@guid()'
+            type: 'Expression'
+          }
+        }
+      }
+      { // Set blob base path
+        name: 'Set Blob Base Path'
+        type: 'SetVariable'
+        dependsOn: [
+          {
+            activity: 'Set Blob Timestamp'
+            dependencyConditions: ['Succeeded']
+          }
+          {
+            activity: 'Set Instance Id'
+            dependencyConditions: ['Succeeded']
+          }
+        ]  
+        policy: {
+          secureOutput: false
+          secureInput: false
+        }
+        userProperties: []
+        typeProperties: {
+          variableName: 'blobBasePath'
+          value: {
+            value: '@concat(\'${recommendationsDataSet}/\', variables(\'blobExportTimestamp\'), \'${recommendationsScope}/\', variables(\'instanceId\'), \'${ingestionIdFileNameSeparator}\')'
+            type: 'Expression'
+          }
+        }
+      }
+      { // Set schema filename        
+        name: 'Set Schema Filename'
+        type: 'SetVariable'
+        dependsOn: []
+        policy: {
+          secureOutput: false
+          secureInput: false
+        }
+        userProperties: []
+        typeProperties: {
+          variableName: 'schemaFile'
+          value: 'recommendations_1.0.json'
```

Comment:
This should be dynamic based on the target data type.

Reply:
Solved